File: sdks/php/src/Resources/GiftCardsResource.php
<?php

namespace Puxbay\Resources;

use Puxbay\Models\GiftCard;
use Puxbay\Models\PaginatedResponse;

class GiftCardsResource extends BaseResource
{
    public function list(int $page = 1, ?string $status = null): PaginatedResponse
    {
        $endpoint = "gift-cards/?page={$page}";
        if ($status) {
            $endpoint .= "&status={$status}";
        }
        $data = $this->get($endpoint);
        return PaginatedResponse::fromArray($data, fn($item) => GiftCard::fromArray($item));
    }

    public function getCard(string $cardId): GiftCard
    {
        $data = $this->get("gift-cards/{$cardId}/");
        return GiftCard::fromArray($data);
    }

    public function create(GiftCard $card): GiftCard
    {
        $data = $this->post('gift-cards/', $card->toArray());
        return GiftCard::fromArray($data);
    }

    public function redeem(string $cardId, float $amount): GiftCard
    {
        $data = $this->post("gift-cards/{$cardId}/redeem/", ['amount' => $amount]);
        return GiftCard::fromArray($data);
    }

    public function checkBalance(string $code): array
    {
        // Balance check might return just {balance: 100} or full card object
        // Return array for flexibility as per Dart implementation
        return $this->get("gift-cards/check-balance/?code={$code}");
    }
}

File: sdks/dotnet/src/Resources/GiftCardsResource.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Puxbay.SDK.Models;

namespace Puxbay.SDK.Resources
{
    public class GiftCardsResource
    {
        private readonly PuxbayClient _client;
        public GiftCardsResource(PuxbayClient client) { _client = client; }

        public async Task<PaginatedResponse<GiftCard>> ListAsync(int page = 1, string status = null)
        {
            string query = $"gift-cards/?page={page}";
            if (!string.IsNullOrEmpty(status)) query += $"&status={status}";
            return await _client.GetAsync<PaginatedResponse<GiftCard>>(query);
        }

        public async Task<GiftCard> GetAsync(string cardId)
        {
            return await _client.GetAsync<GiftCard>($"gift-cards/{cardId}/");
        }

        public async Task<GiftCard> CreateAsync(GiftCard card)
        {
            return await _client.PostAsync<GiftCard>("gift-cards/", card);
        }

        public async Task<GiftCard> RedeemAsync(string cardId, double amount)
        {
            return await _client.PostAsync<GiftCard>($"gift-cards/{cardId}/redeem/", new { amount });
        }
        
        public async Task<Dictionary<string, double>> CheckBalanceAsync(string code)
        {
            return await _client.GetAsync<Dictionary<string, double>>($"gift-cards/check-balance/?code={code}");
        }
    }
}
